### PR TITLE
Add AuTest for HTTP/2 Graceful Shutdown

### PR DESCRIPTION
**Pull request by masaori335**
Add a basic test case of HTTP/2 Graceful Shutdown. Please note that this doesn't cover the issues we discussed in #7267.

### CODE REVIEW

**Review by zizhong — APPROVED** [review]
Looks good

File: tests/gold_tests/h2/gold/nghttp_ts_stderr.gold [hunk 1]
```diff
@@ -0,0 +1,5 @@
+``
+``session free
+``
+``session free
```
Comment (maskit) [inline tests/gold_tests/h2/gold/nghttp_ts_stderr.gold:4]: Just to confirm, there are two because we have two test cases, right?

**Review by maskit — COMMENTED** [review]
Having one gold file for different tests may be hard to maintain.

**Review by shinrich — APPROVED** [review]

**Comment by masaori335** [comment]
Yep, the gold file is assuming 2 test cases are finished before it is checked. I agree it's an ugly hack and increases maintenance cost a bit.
But I can't find any other good approaches. I tried getting metrics via `traffic_ctl`, but it's tricky because ATS needs time to provide stats. You can see how it's tricky on other AuTests ( -e.g. `openclose.test.py`).

**Pull request merged**